modules/amqp091/scanner.go

```diff
@@ -142,6 +144,14 @@ func (scanner *Scanner) Init(flags zgrab2.ScanFlags) error {
 	}
 
 	scanner.config = f
+	scanner.dialerGroupConfig = &zgrab2.DialerGroupConfig{
+		L4TransportProtocol: zgrab2.TransportTCP,
+		BaseFlags:           &f.BaseFlags,
+	}
+	if f.UseTLS {
+		scanner.dialerGroupConfig.TLSFlags = &f.TLSFlags
+		scanner.dialerGroupConfig.TLSEnabled = true
```

Comment:
This is another example where we are setting TLSEnabled slightly differently than banner.
I wonder if we should just have some default "TLS Optional if enabled" dialer group for consistency, and then most of the configuration is in cases where NeedSeparateL4Dialer is true.